node-tests/integration/commands/electron-test.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const ElectronCommand = require('../../../lib/commands/electron');
+const mockElectronProject = require('../../helpers/mock-electron-project');
+const MockUI = require('console-ui/mock');
+const MockAnalytics = require('ember-cli/tests/helpers/mock-analytics');
+const MockProject = require('ember-cli/tests/helpers/mock-project');
+const MockProcess = require('ember-cli/tests/helpers/mock-process');
+const willInterruptProcess = require('ember-cli/lib/utilities/will-interrupt-process');
+const { api } = require('@electron-forge/core');
+const { expect } = require('chai');
+const Builder = require('ember-cli/lib/models/builder');
+const Watcher = require('ember-cli/lib/models/watcher');
+const EventEmitter = require('events');
+const path = require('path');
+const rimraf = require('rimraf');
+const sinon = require('sinon');
+
+class MockWatcher extends EventEmitter {
+  then = sinon.stub().callsFake((resolve) => resolve())
+}
+
+describe('electron command', function() {
+  mockElectronProject();
+
+  let processArgv;
+
+  let command;
+
+  let mockBrocBuilder;
+  let mockBrocWatcher;
+  let mockProcess;
+
+  let createBuilderStub;
+  let cleanupBuilderStub;
+  let createWatcherStub;
+
+  let emitExitStub;
+
+  beforeEach(function() {
+    processArgv = Array.from(process.argv);
+
+    mockBrocBuilder = {};
+    mockBrocWatcher = new MockWatcher();
+    mockProcess = new MockProcess();
+
+    emitExitStub = sinon.stub();
+
+    createBuilderStub = sinon.stub(Builder.prototype, 'setupBroccoliBuilder').callsFake(function() {
+      this.builder = mockBrocBuilder;
+    });
+    sinon.stub(Builder.prototype, 'build').resolves();
+    cleanupBuilderStub = sinon.stub(Builder.prototype, 'cleanup').resolves();
+    createWatcherStub = sinon.stub(Watcher.prototype, 'constructWatcher').returns(mockBrocWatcher);
+    sinon.stub(api, 'start').callsFake(() => {
+      // make electron process exit right after start promise resolves
+      setTimeout(() => {
+        // We call this so we can use sinon's calledAfter() to make sure the
+        // builder's cleanup call came after we emitted the exit event
+        emitExitStub();
+        mockProcess.emit('exit')
+      }, 1);
+      return Promise.resolve(mockProcess);
+    });
+    
+    command = new ElectronCommand({
+      ui: new MockUI(),
+      analytics: new MockAnalytics(),
+      settings: {},
+      project: new MockProject(),
+      tasks: {},
+    });
+
+    // This needs to be initialized when we create Builder instances or it
+    // throws an error
+    willInterruptProcess.capture(mockProcess);
+  });
+
+  afterEach(function() {
+    process.argv = processArgv;
+    willInterruptProcess.release();
+  });
+ 
+  it('works', async function() {
```

Comment:
Not to be hypocritically nit-picky (as I've contributed exactly zero lines of code to this PR), but the other test names are quite descriptive, so seeing a few "it works" tests surprised me. Is it standard/conventional to use a generic name like this for "smoke tests" or were these just overlooked?

Reply:
I've seen this pattern pretty commonly -- when a test module is testing a very specific subject (in this case the `ember electron` command), an `it works` test that tests the basic functionality of the subject, followed by a bunch of more specific tests. We didn't have one of these before because we didn't have a basic functionality tests, only tests of specific conditions, which IMO is a bit of a testing faux pas.

Reply:
(also, not hypocritical -- coding is only part of the story -- review and feedback is vital to making changes, especially big ones like this, so thank you for taking the time!)